crates/telemetry/src/env.rs

```diff
@@ -1,18 +1,40 @@
-/// Returns a boolean indicating if the OTEL layer should be enabled.
+use std::env::VarError;
+
+use opentelemetry_otlp::{
+    OTEL_EXPORTER_OTLP_ENDPOINT, OTEL_EXPORTER_OTLP_METRICS_ENDPOINT, OTEL_EXPORTER_OTLP_PROTOCOL,
+    OTEL_EXPORTER_OTLP_TRACES_ENDPOINT,
+};
+
+/// Returns a boolean indicating if the OTEL tracing layer should be enabled.
 ///
 /// It is considered enabled if any of the following environment variables are set and not empty:
 /// - `OTEL_EXPORTER_OTLP_ENDPOINT`
 /// - `OTEL_EXPORTER_OTLP_TRACES_ENDPOINT`
+///
+/// Note that this is overridden if OTEL_SDK_DISABLED is set and not empty.
+pub(crate) fn otel_tracing_enabled() -> bool {
+    otel_enabled(&[
+        OTEL_EXPORTER_OTLP_ENDPOINT,
+        OTEL_EXPORTER_OTLP_TRACES_ENDPOINT,
+    ]) && !otel_sdk_disabled()
+}
+
+/// Returns a boolean indicating if the OTEL metrics layer should be enabled.
+///
+/// It is considered enabled if any of the following environment variables are set and not empty:
+/// - `OTEL_EXPORTER_OTLP_ENDPOINT`
 /// - `OTEL_EXPORTER_OTLP_METRICS_ENDPOINT`
 ///
 /// Note that this is overridden if OTEL_SDK_DISABLED is set and not empty.
-pub(crate) fn otel_enabled() -> bool {
-    const ENABLING_VARS: &[&str] = &[
-        "OTEL_EXPORTER_OTLP_ENDPOINT",
-        "OTEL_EXPORTER_OTLP_TRACES_ENDPOINT",
-        "OTEL_EXPORTER_OTLP_METRICS_ENDPOINT",
-    ];
-    ENABLING_VARS
+pub(crate) fn otel_metrics_enabled() -> bool {
+    otel_enabled(&[
+        OTEL_EXPORTER_OTLP_ENDPOINT,
+        OTEL_EXPORTER_OTLP_METRICS_ENDPOINT,
+    ]) && !otel_sdk_disabled()
+}
+
+fn otel_enabled(enabling_vars: &[&str]) -> bool {
```

Comment:
Micro-nit: I would name this function by what it does, e.g. `any_vars_set` or something.